clients/java/src/main/java/io/camunda/zeebe/client/impl/command/MigrateProcessInstanceCommandImpl.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.impl.command;
+
+import io.camunda.zeebe.client.api.ZeebeFuture;
+import io.camunda.zeebe.client.api.command.FinalCommandStep;
+import io.camunda.zeebe.client.api.command.MigrateProcessInstanceCommandStep1;
+import io.camunda.zeebe.client.api.command.MigrateProcessInstanceCommandStep1.MigrateProcessInstanceCommandFinalStep;
+import io.camunda.zeebe.client.api.command.MigrationPlan;
+import io.camunda.zeebe.client.api.response.MigrateProcessInstanceResponse;
+import java.time.Duration;
+
+public class MigrateProcessInstanceCommandImpl
```

Comment:
🔧 To avoid users extending this class, let's make it final

While we don't have it automated, this is also [a code convention that we try to apply](https://github.com/camunda/zeebe/wiki/Code-Style#favor-final-modifier).

Reply:
Exactly, I missed this. I also updated `MigrationPlanBuilderImpl`, `MigrationPlanImpl` and `MigrateProcessInstanceResponseImpl` since we want to favour composition over inheritance. 👍